### ISSUE TITLE
[BUG] Azure Monitor OpenTelemetry Exporter client not exporting to Azure

### ISSUE THREAD

**Issue by tyocum17** [Azure/azure-sdk-for-java#32421]
I have instrumented my Spring Boot application to export OpenTelemetry traces/metrics to an Azure Application Insights resource following the documentation here: https://learn.microsoft.com/en-us/java/api/overview/azure/monitor-opentelemetry-exporter-readme?view=azure-java-preview

I am providing the full Connection String from the resource which looks like this:
InstrumentationKey=XXXXXX-XXX-XXXX-XXX-XXX;IngestionEndpoint=https://eastus2-3.in.applicationinsights.azure.com/;LiveEndpoint=https://eastus2.livediagnostics.monitor.azure.com/

This is my code snippet to configure the AzureMonitorExporterBuilder:
```
final AzureMonitorExporterBuilder azureMonitorExporterBuilder = new AzureMonitorExporterBuilder().connectionString("InstrumentationKey=<above instumentation key>;IngestionEndpoint=https://eastus2-3.in.applicationinsights.azure.com/;LiveEndpoint=https://eastus2.livediagnostics.monitor.azure.com/");
```

However, when the actual exporting is taking place I see this URL being created:
https://dc.services.visualstudio.com/v2.1/track

Debugging the code in my IDE I believe this is happening because of the following:

The AzureMonitorExporterBuilder.buildTraceExporter() method internally creates a SpanDataMapper.

https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorExporterBuilder.java#L228

The 2nd parameter of the SpanDataMapper's constructor is a BiConsumer<AbstractTelemetryBuilder, Resource> telemetryInitializer, which references the `populateDefaults` method in the `AzureMonitorExporterBuilder`.

This is creating a new connectionString using only the `instrumentationKey` and not the other properties of the full connectionString passed into the AzureMonitorExporterBuilder. The other properties (e.g. `IngestionEndpoint`) get populated by defaults in `ConnectionStringBuilder` and the new ConnectionString object ends up being cached with the key being the InstrumentationKey and value containing the InstrumentationKey and the incorrect defaults.

The `TelemetryItem` that ends up being exported contains a reference to the InstrumentationKey and when the `TelemetryPipeline.send` method is being called it creates the URL from the cached ConnectionString and then tries to export to the incorrect URL.

I believe the code change required would be to create a connectionString where the key is the instrumentationKey and the value is all of the connection properties provided by the AzureMonitorExporterBuilder and should never reference the defaults in `ConnectionStringBuilder`.
https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/configuration/ConnectionStringBuilder.java#L30-L32

**Comment by joshfree** [comment]
@trask @lmolkova could you please follow up

**Comment by trask** [comment]
hi @tyocum17, thanks for reporting this!

I've sent a PR that (among other things) fixes this: #32454

**Comment by trask** [comment]
this is fixed in the latest release
